tamr_unify_client/operation.py

```diff
@@ -18,6 +18,52 @@ class Operation(BaseResource):
     def from_json(cls, client, resource_json, api_path=None):
         return super().from_data(client, resource_json, api_path)
 
+    @classmethod
+    def from_response(cls, client, response):
+        """
+        Handle idiosyncrasies in constructing Operations from Tamr responses.
+        When a Tamr API call would start an operation, but all results that would be
+        produced by that operation are already up-to-date, Tamr returns:
+            HTTP 204 No Content
+
+        To make it easy for client code to handle these API responses without checking
+        the response code, this method will either construct an Operation, or a
+        dummy NoOp operation representing the fact the 204 Success response.
+
+        :param :class:`~tamr_unify_client.Client` client: Delegate underlying API calls to this client.
+        :param :class:`requests.Response` response: HTTP Response from the request that started the operation.
+        :return: Operation
+        :rtype: :class:`~tamr_unify_client.operation.Operation`
+        """
+        if response.status_code == 204:
+            # Operation was successful, but the response contains no content.
+            # Create a dummy operation to represent this.
+            resource_json = {
+                "id": "-1",
+                "type": "NOOP",
+                "description": "NoOp",
+                "status": {
+                    "state": "SUCCEEDED",
+                    "startTime": "0000-00-00T00:00:00.000Z",
```

Comment:
It might be better to define the time string above as it's repeated several times